### ISSUE TITLE
License and name confusion

### ISSUE THREAD

**Issue by matkoniecz** [Zaczero/openstreetmap-ng#1]
The project readme contains "OpenStreetMap-NG is an unofficial Python fork of openstreetmap.org"
As I understand, project is a rewrite of existing website with code being based on inspecting existing code (not a clean room reimplementation).
Note that https://github.com/openstreetmap/openstreetmap-website/blob/master/LICENSE license is GNU General Public License v2.0.
In such case, why https://github.com/Zaczero/openstreetmap-ng/blob/main/LICENSE is GNU Affero General Public License v3.0 ?
AFAIK you cannot license do derivative work of code licensed only as GPLv2 and release it as AGPLv3.
Also, there is some confusion regarding name - some people got confused (see Weekly OSM) into thinking this project is endorsed or likely replacement of https://github.com/openstreetmap/openstreetmap-website code.
Can you consider changing name into more clear one to make more clear that it is rather a personal experiment?

(note: there was previous contact attempt in PM)

**Comment by Zaczero** [comment]
Hey there!

OpenStreetMap-NG aims to improve across many aspects of the project, one of which is ensuring proper licensing terms. Using GPL-2 for a website is a major misunderstanding of the license, as it provides no protection over network use, such as accessing the application through a web browser (which is the intended mean of distribution). Behind the scenes, I am working on this topic, but due to a heavy workload, it's not currently the top priority. Please stay tuned to the project updates if you want to learn more about the licensing progress. This, along with other issues, will be addressed before the stable release of my work. For your interest, there are other licensing issues within the original project that will also be fixed in OSM-NG, such as using assets with incompatible licenses and redistributing them under GPL. Today, it's not possible to fork the original website and not find yourself in license violations.

Regarding the confusion issue, it might be specific to Weekly OSM due to not providing enough context. This argument could realistically be made against any project using "openstreetmap" as part of its name. Nevertheless, the OSM [trademark policy](https://osmfoundation.org/wiki/Trademark_Policy) clearly permits such use. All resources provided by me explicitly state that they are independent works and not official in any way. I can't force others to do the same.

If you provide me with the specific OSM Weekly edition number that caused confusion, I could help with the investigation and suggest potential changes.

-Kamil

(note: the delay has been caused by my absence - if you are interested in knowing the specifics, I wrote about it in my diaries)

**Comment by matkoniecz** [comment]
Maybe GPL-2 was mistake for original website, maybe not (there are also other concerns beyond mentioned here).

But either way - fork or other derivative work cannot relicense upstream. Even if original license is problematic. See say https://www.gnu.org/licenses/license-list.en.html#AGPLv3.0 (I assume that your work is a derivative work of the original project and you are closely inspired by existing code - I base it on self-description it as a fork and no mentions of a clean room reimplementation)

**Comment by matkoniecz** [comment]
> If you provide me with the specific OSM Weekly edition number that caused confusion, I could help with the investigation and suggest potential changes.

https://duckduckgo.com/?t=ffab&q=openstreetmap-ng+site%3Aweeklyosm.eu&ia=web found https://weeklyosm.eu/archives/17115 and https://weeklyosm.eu/archives/17170 where due to confusing name and no clear description of project status confusion is likely if someone is not already familiar with the situation

**Comment by Zaczero** [comment]
Yup, I see. I think the best course of action would be to make the unofficial notice stand out in my OSM diaries. This would help avoid confusion when people are just skimming through the text and not reading it fully - and especially if they are reading about osm-ng for the first time.

**Comment by Zaczero** [comment]
Here's the wiki page on the relicensing: https://github.com/Zaczero/openstreetmap-ng/wiki/Relicensing I am still working on the dual-licensing agreements and will hopefully resolve it soon!

**Comment by nemobis** [comment]
> I am still working on the dual-licensing agreements and will hopefully resolve it soon!

The openstreetmap-website repository has over 100 contributors, not counting hundreds of translatewiki.net translators. It would probably be easier to convince the project to switch to GPLv2-or-later (if it didn't yet), to simplify future transitions.

**Comment by Zaczero** [comment]
Hello! This project used only a small portion of the previous codebase (with permission), there's no need to ask for permission people who's work is not used. Significant majority of the work is made from scratch. Translatewiki translations remain licensed under the existing translatewiki terms (CC BY 3.0).
